okio/js/src/main/kotlin/okio/Buffer.kt

```diff
@@ -15,147 +15,187 @@
  */
 package okio
 
+import okio.internal.commonCopyTo
+import okio.internal.commonGet
+import okio.internal.commonRead
+import okio.internal.commonReadByte
+import okio.internal.commonReadByteArray
+import okio.internal.commonReadByteString
+import okio.internal.commonReadFully
+import okio.internal.commonWritableSegment
+import okio.internal.commonWrite
+
 actual class Buffer : BufferedSource, BufferedSink {
-  actual override val buffer: Buffer get() = this
+  internal actual var head: Segment? = null
+
+  actual var size: Long = 0L
+    internal set
+
+  actual override val buffer get() = this
 
-  actual override fun emitCompleteSegments(): Buffer = this // Nowhere to emit to!
+  actual override fun emitCompleteSegments() = this // Nowhere to emit to!
 
-  actual override fun emit(): Buffer = this // Nowhere to emit to!
+  actual override fun emit() = this // Nowhere to emit to!
 
-  override fun exhausted(): Boolean = throw UnsupportedOperationException()
+  actual override fun exhausted() = size == 0L
 
-  override fun require(byteCount: Long) {
-    throw UnsupportedOperationException()
+  actual override fun require(byteCount: Long) {
+    if (size < byteCount) throw EOFException()
   }
 
-  override fun request(byteCount: Long): Boolean = throw UnsupportedOperationException()
+  actual override fun request(byteCount: Long): Boolean = size >= byteCount
 
-  override fun peek(): BufferedSource = throw UnsupportedOperationException()
+  actual override fun peek(): BufferedSource {
```

Comment:
These first few methods aren't covered by the common test, should look for more to move over.

Reply:
Have to happen in follow up, and I don't think we'll miss it.